tests/test_jobs.py

```diff
@@ -109,6 +144,24 @@ def test_launch_exec(self, _mock_cluster_state, monkeypatch):
         task = sky.Task(run='echo hi')
         sky.exec(task, cluster_name='test-cluster3', dryrun=True)
 
+        task = sky.Task(run='echo hi')
+        task.set_resources(
+            sky.Resources(disk_tier=resources_utils.DiskTier.BEST))
```

Comment:
Should we add some test for `Azure` specifically?

Reply:
This only tests launching on existing resources, so Azure has nothing special here; we do have some Azure smoke tests that are related to the failover in `test_smoke.py`.

Reply:
We should add some tests that set both `cloud` and `DiskTier` in resources. Different combination of the `cloud` and `disk_tier` for creating `Resources` should be able to be tested without access to the credentials, e.g. `RunPod` with `BEST` should success, while `RunPod` with `LOW` should fail due to unsupported error, and only specifying `BEST` should show all the clouds in the candidate after optimization while `HIGH` should only show part of them.